### PR TITLE
Continue watching correctly after skipping files

### PR DESCRIPTION
**Pull request by dividedmind**
Fix a bug that would prevent watch scanning from working correctly once a couple files has been skipped (perhaps because they have been already up-to-date).

The problem was that `scan()` function had been modified to take and call a callback, but that change had omitted several early returns; in those cases, the callback would never have been called and the jobs would never finish and let the queue continue.

@ahtrotta , could you verify that this still resolves whatever issue #807 fixed?

### CODE REVIEW

**Comment by dividedmind** [comment]
Note to self: retry build at https://github.com/getappmap/vscode-appland/pull/538 after this is merged.

**Review by kgilpin — APPROVED** [review]

**Comment by kgilpin** [comment]
@dustinbyrne you should look at this too because I looked at #807 and I don't understand it.

File: packages/scanner/src/cli/scan/watchScan.ts [hunk 1]
```diff
@@ -111,14 +112,16 @@ export class Watcher {
     );
   }
 
-  private queue = queue<string>(this.scan.bind(this), 2);
+  // do not remove callbackify, apparently on windows
+  // passing plain async function doesn't work (?)
+  private queue = queue<string>(callbackify(this.scan.bind(this)), 2);
```
Comment (dustinbyrne) [inline packages/scanner/src/cli/scan/watchScan.ts:117]: Do you know why this is?

**Review by dustinbyrne — APPROVED** [review]

**Review by ahtrotta — APPROVED** [review]
I just tested and this seems to work fine on Windows.

File: packages/scanner/src/cli/scan/watchScan.ts (hunk 1 above)
Comment (ahtrotta) [reply packages/scanner/src/cli/scan/watchScan.ts:117]: After some digging around, I noticed that only the first N items were being processed on Windows, where N is the concurrency (in our case 2). I googled around and found this [SO post](https://stackoverflow.com/questions/57448099/async-queue-crashes-after-processing-3-initial-elements-of-the-queue-with-concur), and then I added the `callback` to the worker and then all of the items were processed.

**Comment by kgilpin** [comment]
> I just tested and this seems to work fine on Windows.

@ahtrotta can you resolve the merge conflict and merge it?

**Pull request merged**